File: PAM_Python/imp_exp.py
class ImpExp:
    def __init__(self):
        self.d = None  # Matriz de disimilitud
        self.nameObjects = None  # Nombres de los objetos
        self.nObjects = 0  # Número de objetos
        self.nVariables = 0  # Número de variables

    # Importa la matriz de costos desde un archivo
    def importMatrixCost(self, fileName):
        with open(fileName, 'r') as file:
            lines = file.readlines()
            self.nObjects = len(lines) - 1
            self.nVariables = self.nObjects
            self.nameObjects = [''] * self.nObjects
            self.d = [[0.0 for _ in range(self.nObjects)] for _ in range(self.nObjects)]

            # Saltar la primera línea (encabezado)
            for i in range(self.nObjects):
                line = lines[i + 1].strip()
                if not line:
                    raise IOError(f"Línea vacía en el archivo en la posición {i + 1}")
                parts = line.split()
                self.nameObjects[i] = parts[0]
                for j in range(self.nObjects):
                    if len(parts) <= j + 1 or not parts[j + 1].strip():
                        raise IOError(
                            f"Valor faltante en la matriz en la fila {i + 1}, columna {j + 1}")
                    self.d[i][j] = float(parts[j + 1])

    # Obtiene el número de filas en un archivo
    def getNRows(self, fileName):
        with open(fileName, 'r') as file:
            return sum(1 for _ in file)

File: PAM_Java/ImpExp.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class ImpExp {

    public float[][] d; // Matriz de disimilitud
    public String[] nameObjects; // Nombres de los objetos
    public int nObjects; // Número de objetos
    public int nVariables; // Número de variables

    // Importa la matriz de costos desde un archivo
    public void importMatrixCost(String fileName) throws IOException {
        try (BufferedReader br = new BufferedReader(new FileReader(fileName))) {
            nObjects = getNRows(fileName) - 1;
            nVariables = nObjects;
            nameObjects = new String[nObjects];
            d = new float[nObjects][nObjects];

            String line;
            br.readLine(); // Saltar la primera línea (encabezado)
            for (int i = 0; i < nObjects; i++) {
                line = br.readLine();
                if (line == null || line.trim().isEmpty()) {
                    throw new IOException("Línea vacía en el archivo en la posición " + (i + 1));
                }
                String[] parts = line.split("\\s+");
                nameObjects[i] = parts[0];
                for (int j = 0; j < nObjects; j++) {
                    if (parts.length <= j + 1 || parts[j + 1].trim().isEmpty()) {
                        throw new IOException(
                                "Valor faltante en la matriz en la fila " + (i + 1) + ", columna " + (j + 1));
                    }
                    d[i][j] = Float.parseFloat(parts[j + 1]);
                }
            }
        }
    }

    // Obtiene el número de filas en un archivo
    public int getNRows(String fileName) throws IOException {
        int count = 0;
        try (BufferedReader br = new BufferedReader(new FileReader(fileName))) {
            while (br.readLine() != null) {
                count++;
            }
        }
        return count;
    }
}